src/reducers/targetReducer.js

```diff
@@ -5,34 +5,22 @@ const targetReducer = (state = mapInitialState, action) => {
   switch (action.type) {
     case types.START_NEW_TARGET: {
       return state.set('addingNewTarget', true)
-        .set('newTargetLat', action.payload.lat)
-        .set('newTargetLong', action.payload.lng);
+        .set('newTarget', action.newTarget);
     }
     case types.END_NEW_TARGET: {
       return state.set('addingNewTarget', false)
-        .set('targetRadius', mapInitialState.get('targetRadius'))
         .set('targetErrors', {});
     }
-    case types.UPDATE_RADIUS: {
-      return state.set('targetRadius', action.payload);
-    }
-    case types.CREATE_TARGET: {
-      return state.set('targetRadius', action.payload);
-    }
     case types.CREATE_TARGET_SUCCESS: {
-      return state.update('targets', targetList => targetList.push(action.payload.target))
+      return state.update('targets', targetList => targetList.push(action.target))
         .set('targetErrors', {})
-        .set('addingNewTarget', false)
-        .set('targetRadius', mapInitialState.get('targetRadius'));
+        .set('addingNewTarget', false);
     }
-    case types.CREATE_TARGET_ERROR: {
-      return state.set('targetErrors', action.payload.errors);
+    case types.API_ERROR: {
+      return state.set('targetErrors', action.errors);
```

Comment:
when are you clearing this error? Having a generic action `API_ERROR` will trigger in all reducers and you might end up with unwanted behavior.

IMO a much cleaner way to handle these kind of errors is to throw a `SubmissionError` directly that will be caught by redux-form for the form that was just submitted, it saves you the hustle of passing to the reducer and getting it in the corresponding form.
It also lets you do neat things like if the backend returns an error for a specific field, it will appear as a validation error on that field.

This is the code I usually use:
```
catch ({ errors, error }) {
  throw new SubmissionError(typeof errors === 'object' ? errors : { _error: errors || error });
}
```
`_error` is the parameter for the global error in a form. Then you just get it as a prop called `error` in the form.

Here is the documentation https://redux-form.com/8.1.0/docs/api/submissionerror.md/.

In the case that you have a loading indicator, you would still have to dispatch an action to put the loader in `false`.

Reply:
Agree with this one as it is the redux-form's way of doing it 👍

Reply:
@aguscha333 I've implemented SubmissionError for the `createTarget` submission but I'm still using the `ApiError` action to handle errors while calling the `/GetTopics` endpoint since it's not submitting a form.

Reply:
Great! I still think that you should should decentralize the error action and have one per reducer so it's easier to keep it in control. As you have it right now if you have in two reducers a case for `API_ERROR` you will be setting the error in both reducers.